### PR TITLE
Config tweak to allow for easier local app+api

### PR DESCRIPTION
**Pull request by martyn-w**
Update to allow for local api
Update docs

### Trello card
[Update developer docs](https://trello.com/c/XhievTyj/5923-review-and-update-developer-documentation-for-website)

### Context
Small change to allow for easier development with a local API

### Changes proposed in this pull request
Allow for a local API to be sent via ENV VARs
Update docs

### Pre-election period restrictions

* [x] This change is permitted within the constraints of the [pre-election period guidelines](https://www.gov.uk/government/publications/election-guidance-for-civil-servants/general-election-guidance-2024-guidance-for-civil-servants-html#publishing-content-online-)

### CODE REVIEW

**Review by asatwal — DISMISSED** [review]
👍 LGTM

File: config/initializers/git_api_client.rb [hunk 1]
```diff
@@ -7,7 +7,18 @@
   if endpoint
     parsed = URI.parse(endpoint)
 
-    config.host = parsed.hostname
+    # if using a non-standard port, include it as part of the host setting
+    config.host = if parsed.port != 443
+                    "#{parsed.hostname}:#{parsed.port}"
+                  else
+                    parsed.hostname
+                  end
```
Comment (carlosmartinez) [inline config/initializers/git_api_client.rb:15]:
```suggestion
    config.host = parsed.port == 443 ? parsed.hostname : "#{parsed.hostname}:#{parsed.port}"
```

File: config/initializers/git_api_client.rb [hunk 2]
```diff
@@ -7,7 +7,18 @@
...
+
+    # if using a local API for dev, don't verify self-certified ssl certs
+    if Rails.env.development? && parsed.hostname == "localhost" && parsed.scheme == "https"
```
Comment (carlosmartinez) [inline config/initializers/git_api_client.rb:18]: suggest extracting `Rails.env.development? && parsed.hostname == "localhost" && parsed.scheme == "https"` to a method `local_dev?` or similar.

File: docs/development.md [hunk 3]
```diff
@@ -9,6 +9,17 @@
 * Run `bundle exec rails server` to launch the app on http://localhost:3000
 * Run `./bin/shakapacker-dev-server` in a separate shell for faster compilation of assets
 
+### Running the app with a local api
+
+* Build the [get-into-teaching-api](https://github.com/DFE-Digital/get-into-teaching-api) following the documentation in that project. Configure it use a dev CRM instance and docker-based Redis and Postgresql database
+* Run the api on https://localhost:5001/api
```
Comment (carlosmartinez) [inline docs/development.md:15]:
```suggestion
* Run the API on https://localhost:5001/api
```

File: docs/development.md [hunk 4]
```diff
@@ -9,6 +9,17 @@
...
+* Run the app using env vars to point to the local api:
```
Comment (carlosmartinez) [inline docs/development.md:16]:
```suggestion
* Run the app using env vars to point to the local API:
```

**Review by carlosmartinez — DISMISSED** [review]

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**